src/sources.rs

```diff
@@ -92,8 +96,21 @@ impl Iter {
         let mut children = Vec::<(OsString, bool)>::new();
         for entry in readdir {
             let entry = try!(entry);
+            let entry_name = entry.file_name();
             let ft = try!(entry.file_type());
-            children.push((entry.file_name(), ft.is_dir()));
+
+            if self.excludes.is_some() &&
+                self.excludes.clone().unwrap().matches(&entry_name).len() > 0 {
```

Comment:
I don't think you should need to clone it?

I think you can just use `is_match` rather than checking the len?